### ISSUE TITLE
loss and gradient backpropagation

### ISSUE THREAD

**Issue by lk0429** [linyq2117/CLIP-ES#23]
Sorry to bother you, I didn't find the loss function and gradient backpropagation process while reading the code. I would like to ask if the initial CAM can be obtained directly? No training required? I see that there is a gradient backpropagation process in the model diagram. If the initial CAM is obtained through training, in which part of the code are the loss function and gradient backpropagation process located? Thanks！

**Comment by linyq2117** [comment]
Thanks for your interest!

Our method is based on the pre-trained CLIP model and does not require additional training. We use GradCAM to obtain the CAM, which is based on the gradient of target classes. You can refer to [GradCAM](https://arxiv.org/abs/1610.02391) for its main idea.

**Comment by lk0429** [comment]
不好意思打扰一下，请问：
1.initial CAM是直接通过预训练的CLIP直接得到吗？
2.initial CAM是固定不变的吗？initial CAM通过GradCAM得到，GradCAM反向传播 参数更新的话，initial CAM是会变吧
3.如果我想对initial CAM 做一个监督（比如用一个可靠的mask），希望initial CAM训练的越来越好，那我的反向传播参数更新哪一部分呢？


谢谢您的耐心解答！


李坤
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "linyq2117/CLIP-ES"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年7月16日(星期二) 晚上11:02
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [linyq2117/CLIP-ES] loss and gradient backpropagation (Issue #23)





 
Thanks for your interest!
 
Our method is based on the pre-trained CLIP model and does not require additional training. We use GradCAM to obtain the CAM, which is based on the gradient of target classes. You can refer to GradCAM for its main idea.

**Comment by linyq2117** [comment]
1. 是的。
2. initial CAM是固定不变的。CAM的思想是：对于训练好的分类模型，给定图像和一个目标类别，找到特征图中与这个类别相关的区域，从而生成CAM。因此对于给定的分类模型，CAM一般是确定的。（对于CLIP这种图文模型，可以通过设定不同的text prompt来得到更优的结果）。GradCAM只传播梯度用于计算CAM，不会进行参数更新，您可以参考CAM和GradCAM的文章来了解更多细节。
3. 您可能误解了CAM的概念和用途。CAM是针对分类模型设计的，旨在通过image-level标签训练的分类模型，得到pixel-level的mask。如果已经拥有可靠的mask，可以直接用其训练语义分割模型。

**Comment by lk0429** [comment]
> Sorry to bother you, I didn't find the loss function and gradient backpropagation process while reading the code. I would like to ask if the initial CAM can be obtained directly? No training required? I see that there is a gradient backpropagation process in the model diagram. If the initial CAM is obtained through training, in which part of the code are the loss function and gradient backpropagation process located? Thanks！



> Thanks for your interest!
> 
> Our method is based on the pre-trained CLIP model and does not require additional training. We use GradCAM to obtain the CAM, which is based on the gradient of target classes. You can refer to [GradCAM](https://arxiv.org/abs/1610.02391) for its main idea.
